sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -104,8 +109,10 @@ object FileFormatWriter extends Logging {
       hadoopConf: Configuration,
       partitionColumns: Seq[Attribute],
       bucketSpec: Option[BucketSpec],
+      statsTrackers: Seq[WriteJobStatsTracker],
       refreshFunction: (Seq[ExecutedWriteSummary]) => Unit,
```

Comment:
instead of having the `refreshFunction`, can we just let this `write` method return `Seq[ExecutedWriteSummary]`